tensorzero-core/src/config/mod.rs

```diff
@@ -1199,6 +1251,142 @@ impl Config {
     }
 }
 
+/// This is for privacy of internal types
+pub mod unwritten_config {
```

Comment:
this turned out to be a pub mod - is this intentional (i'm not sure you can get around it since config writing happens outside of this module)? consider moving this into `unwritten_config.rs`​ file?